### PR TITLE
[WebProfilerBundle] Add collapsed sidebar on small screens

### PR DESCRIPTION
**Pull request by hason**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Desktop
Desktop

Mobile
Mobile Mobile touch

### CODE REVIEW

**Comment by javiereguiluz** [comment]
I'm :-1: on this change too. I purposely removed this feature while the redesign.

My reasoning:

  1) The new sidebar is narrower than the old one, so the main contents enjoy more space.
  2) Collapsing the sidebar gives you just a bit more pixels for the content.
  3) Making the lines that contain text longer is not better but worse (as explained in the other pull request)
  4) Lately Symfony favors not adding new features and even removing some of them if possible. I think that this "collapsible sidebar" can be a "removable feature" and even we could get rid of the `position` config. option.

@hason in any case, keep in mind that my vote doesn't count for the merging or not of this PR, so you just have to convince the Symfony deciders.

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig [hunk 1]
```diff
@@ -62,7 +62,22 @@
...
+                    <script type="text/javascript">
+                        (function(Sfjs) {
+                            if ('true' == Sfjs.getPreference('sidebar/collapse')) {
+                                Sfjs.addClass(document.body, 'sidebar-collapse');
+                            }
+                        })(Sfjs);
+                    </script>
                     <ul id="menu-profiler">
+                        <li>
+                            <a href="#" onclick="javascript: Sfjs.toggleClass(document.body, 'sidebar-collapse'); Sfjs.setPreference('sidebar/collapse', Sfjs.hasClass(document.body, 'sidebar-collapse')); return false;">
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig]: ``javascript:`` in the onclick handler is wrong. You are confusing onclick and special href

**Comment by stof** [comment]
I'm not convinced either we need this. I can tell that I never collapsed the profiler menu (I collapse the toolbar sometimes, for instance when it hides some of my own ``position: fixed`` content, but not the menu)

**Comment by stloyd** [comment]
:-1: for collapsing using javascript,
:+1: for collapsing on small screens with expanding on click/on hover.

**Comment by hason** [comment]
The collapsed sidebar is enabled only on small screens.

**Comment by javiereguiluz** [comment]
Now that the behavior of this proposed feature has changed, I change my vote too for :+1: 

There are very small alignment issues:

  * When collapsed, the menu icon doesn't appear centered horizontally as the other icons.
  * When expanded, the icon of the Search button should be centered.
  * When expanded, the menu icon should be vertically centered with the "Latest Profiles" and the Search buttons.

**Comment by hason** [comment]
@javiereguiluz I tried to correct alignment issues, but I'm not a webmaster. Could you review it, please?

**Comment by fabpot** [comment]
Thank you @hason.

**Pull request merged**